### PR TITLE
fix(test): update travis and e2e selfSignedCert fn

### PR DESCRIPTION
**Pull request by dnwe**
- add go.mod/go.sum
- update .travis.yml to test against supported Go versions
- drop golint from CI checks

To satisfy modern Go crypto in the tlsrouter e2e_test.go:
- bump RSA keysize to minimum 2048 bits
- set NotBefore to valid recent timestamp
- include CommonName in SANs

### CODE REVIEW

File: cmd/tlsrouter/e2e_test.go [hunk 1]
```diff
@@ -192,16 +192,16 @@ func selfSignedCert(domains []string) (tls.Certificate, *x509.CertPool, error) {
...
 		},
-		NotBefore:             time.Time{},
+		NotBefore:             time.Now().Add(-5 * time.Minute),
 		NotAfter:              time.Now().Add(60 * time.Minute),
 		IsCA:                  true,
 		KeyUsage:              x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
 		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
 		BasicConstraintsValid: true,
-		DNSNames:              domains[1:],
+		DNSNames:              domains[:],
```
Comment (danderson) [inline cmd/tlsrouter/e2e_test.go:201]: Why this change? Doesn't this map to additional domains, where domains[0] is the primary CN in the cert?

Comment (dnwe) [reply cmd/tlsrouter/e2e_test.go:201]: This is because CommonName has long been marked as deprecated and was always meant to be duplicated in the subject alternative names (along with any additional names). In Go 1.16 the unittests would fail when a single domain was passed to this func because it left the subject alt names completely empty and was only using the common name field. 

Quoting Filo:
> CN has been deprecated for 20 years, the specs do not recommend supporting it anymore, the browsers successfully dropped it, the WebPKI CAs are prohibited from setting it to anything that is not also in the SANs, and its behavior is completely unspecified.

Also Ref: https://go-review.googlesource.com/c/go/+/231379

Comment (dnwe) [reply cmd/tlsrouter/e2e_test.go:201]: i.e.,
```
--- FAIL: TestRouting (0.69s)
    e2e_test.go:82: get "test.com" failed: dial TLS "127.0.0.1:52685" for "test.com": 
    x509: certificate relies on legacy Common Name field, use SANs or temporarily 
    enable Common Name matching with GODEBUG=x509ignoreCN=0
```

**Comment by dnwe** [comment]
I'm guessing the travis.yml changes are somewhat moot as it looks like Travis has been disabled on this repo anyway, but I guess there's no harm in the travis.yml being up-to-date, and then Travis can either be reinstated or converted to use Github actions later

File: cmd/tlsrouter/e2e_test.go (hunk 1 above)
Comment (danderson) [reply cmd/tlsrouter/e2e_test.go:201]: TIL, thanks!

**Comment by danderson** [comment]
Travis went sour on OSS a while back. We should probably update to GH actions, but for now an up-to-date CI config is better than nothing.

**Comment by danderson** [comment]
Thank you!
